satpy/readers/osisaf_l3_nc.py

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2023 Satpy developers
+#
+# This file is part of satpy.
+#
+# satpy is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# satpy is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# satpy.  If not, see <http://www.gnu.org/licenses/>.
+# type: ignore
```

Comment:
@simonrp84 Why is this line here? This is why mypy is failing.

Reply:
```suggestion
```

Reply:
That line was there because I copy/pasted the GHRSST reader as the base for my code...and that reader has the line. Should it be removed there too?

Reply:
Lol yeah, in the long run we should be type checking everything...or at least letting mypy choose what it does. That likely needs its own PR since that ignore is probably there for a reason.